### PR TITLE
Feature : github-actions - Setup stale Issues and PRs workflow

### LINKED ISSUE #689: Feature : github-actions - Setup stale Issues and PRs workflow
**Description:**
This feature involves setting up a workflow to manage stale issues and pull requests (PRs) effectively. Stale issues and PRs can accumulate over time, leading to clutter and reduced productivity. By implementing this workflow, we aim to automatically identify and label stale issues and PRs based on predefined criteria. Additionally, the workflow will incorporate actions such as notifying contributors, closing stale items, or prompting for updates to keep the issue tracker organized and relevant. This feature will streamline issue and PR management, ensuring that active contributions receive timely attention while maintaining a clean and responsive development environment.

Something like this:

Image

### PR DESCRIPTION
**Pull request by aialok**
**What kind of change does this PR introduce?**
- This PR will setup workflow to manage stale issues and pull-request. 

**Issue Number:**

-  Closes #689 

**Screenshots/videos:**

**If relevant, did you update the documentation?**

**Summary**
This feature involves setting up a workflow to manage stale issues and pull requests (PRs) effectively. Stale issues and PRs can accumulate over time, leading to clutter and reduced productivity. By implementing this workflow, we aim to automatically identify and label stale issues and PRs based on predefined criteria. Additionally, the workflow will incorporate actions such as notifying contributors, closing stale items, or prompting for updates to keep the issue tracker organized and relevant. This feature will streamline issue and PR management, ensuring that active contributions receive timely attention while maintaining a clean and responsive development environment.

**Does this PR introduce a breaking change?**

- No

### CODE REVIEW

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+         
+        # Message to comment on pull requests that are about to be closed.
+        close-pr-message: 'This pull request did not get any activity in the past 180 days and thus has been closed.'
+           
+        # Labels to add to stale issues and pull requests.
+        stale-issue-label: 'no-issue-activity'
+        stale-pr-label: 'no-pr-activity'
+        
+        # Number of days of inactivity before an issue/PR is marked as stale.
+        days-before-stale: 14
```
Comment (benjagm) [inline .github/workflows/stale.yml:58]:
```suggestion
        days-before-stale: 30
```

Let's move it to 30 days.

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          Thank you for your patience :heart:
+        
+        # Message to comment on issues that are about to be closed.
+        close-issue-message: 'This issue did not get any activity in the past 180 days and thus has been closed. Please check if the main branch has fixed it. Please, create a new issue if the issue is not fixed.'
+         
+        # Message to comment on pull requests that are about to be closed.
+        close-pr-message: 'This pull request did not get any activity in the past 180 days and thus has been closed.'
+           
+        # Labels to add to stale issues and pull requests.
+        stale-issue-label: 'no-issue-activity'
```
Comment (benjagm) [inline .github/workflows/stale.yml:54]:
```suggestion
        stale-issue-label: 'Status: Stale'
```

File: .github/workflows/stale.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        stale-pr-label: 'no-pr-activity'
```
Comment (benjagm) [inline .github/workflows/stale.yml:55]:
```suggestion
        stale-pr-label: 'Status: Stale'
```

**Review by benjagm — CHANGES_REQUESTED** [review]
Great job Alok. I just left some suggestions.

**Comment by aialok** [comment]
Done with the requested changes.
Thank you  : )

File: .github/workflows/stale.yml [hunk 4]
```diff
@@ -0,0 +1,68 @@
+name: Mark stale issues and pull requests
+
+on:
+  schedule:
+  - cron: "0 0 * * *" # Every day at 00:00 UTC
```
Comment (benjagm) [inline .github/workflows/stale.yml:5]: I think weekly is enough.

**Review by benjagm — CHANGES_REQUESTED** [review]
I left a small comment to change the trigger periodicity to weekly instead daily.

**Comment by aialok** [comment]
I made the requested changes. One test is going to fail authorized file detection one as the changes in the workflow file.

**Review by benjagm — APPROVED** [review]
Looks good. Thanks Alok.

**Review by benjagm — APPROVED** [review]
Good job Alok!

**Pull request merged**